cxx-qt-gen/src/writer/cpp/mod.rs

```diff
@@ -275,12 +292,19 @@ mod tests {
         MyObject::MyObject(QObject* parent)
           : QStringListModel(parent)
           , m_rustObj(cxx_qt::my_object::cxx_qt_my_object::createRs())
+          , m_rustObjMutex(std::make_shared<std::mutex>())
+          , m_cxxQtThreadPtr(std::make_shared<QPointer<MyObject>>(this))
+          , m_cxxQtThreadPtrMutex(std::make_shared<std::shared_mutex>())
         {
           cxx_qt::my_object::cxx_qt_my_object::initialiseCpp(*this);
           m_initialised = true;
         }
 
-        MyObject::~MyObject() = default;
+        MyObject::~MyObject()
+        {
+          const auto guard = std::unique_lock(*m_cxxQtThreadPtrMutex);
```

Comment:
should we also lock the m_rustObjMutex here? To ensure that the C++ object isn't destroyed while we are also somewhere in the Rust side?

Reply:
This sounds like a different problem that would need a different solution. As yes you could block an invokable from using the m_rustObj until the deconstructor has completed, but then what happens on release? The invokable is then trying to call a m_rustObj that doesn't exist in a QObject that doesn't exist?